packages/core/database/src/entity-manager/regular-relations.ts

```diff
@@ -24,8 +24,8 @@ declare module 'knex' {
 }
 
 //  TODO: This is a short term solution, to not steal relations from the same document.
-const getDocumentSiblingIdsQuery = (con: Knex, tableName: string, id: ID) => {
-  // Find if the model is a content type or something else (e.g component)
+const createDocumentSiblingsIDSubQuery = (tableName: string, id: ID, trx?: Knex.Transaction) => {
```

Comment:
@Marc-Roig js changed this code too, that will probably conflict with your change but using the trx looks nice too

Reply:
oops! We can merge the changes

Aren't transactions automatically applied to subqueries? :O

Reply:
I initially thought the same but had a concurrency issue, added transaction support and it worked.

Also asked some AI assistants (Copilot, jetbrain's) and they both said to me that knex sub queries don't inherit from the parent transactions. I'm having a hard time finding official documentation on this. I'll try digging a bit in Knex's codebase to confirm

Reply:
interesting!